### DISCUSSION TITLE
GCP Support (GCS and pub/sub)

### DISCUSSION THREAD

**Discussion by rhysxevans** [opensearch-project/data-prepper#4352] [Ideas]
Hi

Are there any plans to support data being sourced from GCP ? 

I am thinking initially pub/sub and GCS bucket support.  Ideally this would support both native gcp authentication and also cross cloud support via something like workload identity federation.

Any thoughts 

Thanks

**Comment by dlvenable** [comment]
@rhysxevans , I do not know of any particular plans for this. I believe that Google Cloud Storage is compatible with Amazon S3. So the `s3` source could possibly support this if it had an `google_cloud:` configuration similar to the existing `aws:` configuration.

The pub/sub support might be a somewhat larger task. We'd need a new notification_type for Google's solution as another option alongside the existing Amazon SQS support.

Do you see any other gaps to accomplishing this?

We are happy to take community contributions like this. So feel free to open a GitHub issue for it and you can raise a PR. We can also help provide guidance on how to implement.

**Comment by rhysxevans** [comment]
Hi, apologies. I have been away. 

I am not a developer, so not 100% sure of all the gaps. That been said I am trying to get a developer resource to have a look at this along with a few other things. Hopefully I can get some progress on that front soon

Thanks
